### PR TITLE
MacVim, vim: update & use lua PG

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: editors/MacVim/files/patch-remove-updater.diff [hunk 1]
```diff
@@ -14,21 +14,31 @@
...
+@@ -11,13 +11,11 @@
                  <outlet property="advancedPreferences" destination="620" id="632"/>
+                 <outlet property="allowForceClickLookUpButton" destination="rlt-zw-mfW" id="xCv-HS-zyJ"/>
                  <outlet property="appearancePreferences" destination="hr4-G4-3ZG" id="G54-DD-ACh"/>
 -                <outlet property="autoInstallUpdateButton" destination="UYM-W0-Kgl" id="cX5-tk-9WJ"/>
                  <outlet property="generalPreferences" destination="115" id="143"/>
+                 <outlet property="inputPreferences" destination="Bnq-Nx-GJH" id="FES-rQ-Fpa"/>
                  <outlet property="layoutPopUpButton" destination="427" id="596"/>
+                 <outlet property="sparklePrereleaseButton" destination="4Y5-mA-blQ" id="baN-Dn-dgO"/>
+                 <outlet property="sparklePrereleaseDesc" destination="ere-oJ-WLd" id="fTe-GS-5qR"/>
```
Comment (raimue) [inline editors/MacVim/files/patch-remove-updater.diff:26]: These buttons seem related to Sparkle... should we remove them as well?

**Review by raimue — COMMENTED** [review]
Thank you for the port update! The changes to use the lua port group make sense. :+1:

Only one remark regarding the patch to disable the Sparkle updater for MacVim inline.

File: editors/MacVim/files/patch-remove-updater.diff (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply editors/MacVim/files/patch-remove-updater.diff]: I updated the PR to remove `sparklePrereleaseButton` and `sparklePrereleaseDesc`.

**Review by raimue — APPROVED** [review]
Thank you again! Changes look good to me now. :+1:

**Pull request merged**